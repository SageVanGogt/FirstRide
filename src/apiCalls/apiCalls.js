import { geoKey } from './../apiKey';

const signinUser = async (user) => {
  const url = 'http://localhost:3000/api/users/';
  const init = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, init);
  const profile = await response.json();
  return profile;
}; 

const signupUser = async (user) => {
  const url = 'http://localhost:3000/api/users/new';
  const body = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const userRecieved = await response.json();
  return userRecieved;
};

const fetchDestination = async (name) => {
  const url = `http://localhost:3000/api/locations/${name}`;
  const response = await fetch(url);
  const destination = await response.json();
  return destination;
};

const fetchRides = async (destination) => {
  const url = `http://localhost:3000/api/rides/get/${destination}`;
  const body = {
    method: 'POST', 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = await response.json();
  return rides;
};

const fetchPickups = async (location) => {
  const url = `http://localhost:3000/api/pickup/get/${location}`;
  const body = {
    method: "POST", 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const pickups = await response.json();
  return pickups;
};

const fetchGeocode = async (address) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${geoKey}`;  
  try {
    const response = await fetch(url);
    const locationData = await response.json();
    if (!locationData) {
      return new Error('error')
    }
    return locationData;
  } catch (error) {
    throw error;
  }
}; 

const submitNewRide = async (ride) => {
  const url = `http://localhost:3000/api/rides/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(ride),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const rideId = await response.json();
  return rideId;
};

const submitNewPickup = async (location) => {
  const url = `http://localhost:3000/api/pickup/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(location),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const pickupId = await response.json();
  return pickupId;
};

const fetchRidesPassengers = async (locationId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/passengers/${locationId}`;
  const body = {
    method: 'POST', 
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const passengers = await response.json();
  return passengers;
};

const postRidesPassengers = async (newPassenger) => {
  const url = `http://localhost:3000/api/rides_passengers/new`;   
  const body = {
    method: "POST",
    body: JSON.stringify(newPassenger),
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const ridePassenger = await response.json();
  return ridePassenger;
};

const postNewProfile = async (profile) => {
  const url = `http://localhost:3000/api/profiles/new`;
  const body = {
    method: "POST",
    body: JSON.stringify(profile),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const profileInfo = await response.json();
  return profileInfo;
};

const removePassengerRide = async (rideId, passengerId, destinationId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/passengers/${passengerId}/destination/${destinationId}`; 
  const body = {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const deletedId = await response.json();
  return deletedId;
};

const fetchUserRides = async (userId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/rides/${userId}`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = response.json();
  return rides;
};

const fetchRidesFromUser = async (rideId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/get/user/`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const ridesInfo = await response.json();
  return ridesInfo;
};

const reverseGeoCode = async (lat, lng) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${geoKey}`;
  const response = await fetch(url);
  const geoData = await response.json();
  return geoData;
}

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewRide,
  submitNewPickup,
  fetchRidesPassengers,
  postRidesPassengers,
  postNewProfile,
  removePassengerRide,
  fetchUserRides,
  fetchRidesFromUser,
  reverseGeoCode
};