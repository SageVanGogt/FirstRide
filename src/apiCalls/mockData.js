export const mockUser = {
  id:1,
  user_id:"456",
  user_img:"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
  bio:"I like to eat",
  rating:"Solid 7"
}