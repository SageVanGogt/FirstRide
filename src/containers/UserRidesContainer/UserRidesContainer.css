.user-rides-container {
  position: absolute;
  z-index: 4;
  height: 400px;
  width: 600px;
  background-color: white;
}