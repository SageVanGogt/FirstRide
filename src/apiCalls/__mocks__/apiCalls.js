import * as MOCK from './../mockData';

const signinUser = jest.fn().mockImplementation(() => 
  Promise.resolve({
    user: [{
    "id":1,
    "user_id":"456",
    "user_img":"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
    "bio":"I like to eat",
    "rating":"Solid 7"
    }]
  })
);

const signupUser = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1
}));

const fetchDestination = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1,
  "location_name": "Red Rocks",
  "location_lat_lng": "12341, 123412"
}));

const fetchRides = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockRides));

const fetchPickups = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockPickups));

const fetchGeocode = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockGeoInfo));

const submitNewRide = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const submitNewPickup = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const postRidesPassengers = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewPickup,
  submitNewRide,
  postRidesPassengers
}