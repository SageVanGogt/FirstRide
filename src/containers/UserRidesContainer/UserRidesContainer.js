import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as API from './../../apiCalls/apiCalls';
import './UserRidesContainer.css';

export class UserRidesContainer extends Component {
  constructor(props) {
    super(props)
  };

  loadPassengerRides = async (locationId) => {
    const rides
  }

  loadRideDetails = async (rideArray) => {

  }

  render() {
    return (
      <div className="user-rides-container">
        <article>
          <h2>Red Rocks</h2>
          <ul>
            {this.loadPassengerRides(1)}
          </ul>
        </article>
        <article>
          <h2>Breckenridge</h2>
          <ul>
            {this.loadPassengerRides(2)}
          </ul>
        </article>
      </div>
    );
  };
};

export const mapStateToProps = (state) => ({

});

export const mapDispatchToProps = (dispatch) => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(UserRidesContainer);