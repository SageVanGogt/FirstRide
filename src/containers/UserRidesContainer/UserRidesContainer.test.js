import React from 'react';
import { shallow } from 'enzyme';
import { UserRidesContainer } from './UserRidesContainer';
import * as API from './../../apiCalls/apiCalls';

jest.mock('./../../apiCalls/apiCalls');

describe('UserRidesContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<UserRidesContainer />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('loadPassengerRides', () => {
    it('should call fetchRidesPassenger with the correct params', async () => {
      let expected = 1;
      await wrapper.instance().loadPassengerRides();

      expect(API.fetchRidesPassengers).toHaveBeenCalledWith(expected)
    })
  })
})