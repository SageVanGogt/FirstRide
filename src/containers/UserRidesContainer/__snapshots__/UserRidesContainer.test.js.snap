// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRidesContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRidesContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <h2>
                    Red Rocks
          </h2>
          <ul>
                    Promise {}
          </ul>
</article>,
        <article>
          <h2>
                    Breckenridge
          </h2>
          <ul>
                    Promise {}
          </ul>
</article>,
      ],
      "className": "user-rides-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Red Rocks
</h2>,
            <ul>
              Promise {}
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Red Rocks",
            },
            "ref": null,
            "rendered": "Red Rocks",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Promise {},
            },
            "ref": null,
            "rendered": Promise {},
            "type": "ul",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Breckenridge
</h2>,
            <ul>
              Promise {}
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Breckenridge",
            },
            "ref": null,
            "rendered": "Breckenridge",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Promise {},
            },
            "ref": null,
            "rendered": Promise {},
            "type": "ul",
          },
        ],
        "type": "article",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <h2>
                        Red Rocks
            </h2>
            <ul>
                        Promise {}
            </ul>
</article>,
          <article>
            <h2>
                        Breckenridge
            </h2>
            <ul>
                        Promise {}
            </ul>
</article>,
        ],
        "className": "user-rides-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Red Rocks
</h2>,
              <ul>
                Promise {}
</ul>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Red Rocks",
              },
              "ref": null,
              "rendered": "Red Rocks",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Promise {},
              },
              "ref": null,
              "rendered": Promise {},
              "type": "ul",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Breckenridge
</h2>,
              <ul>
                Promise {}
</ul>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Breckenridge",
              },
              "ref": null,
              "rendered": "Breckenridge",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Promise {},
              },
              "ref": null,
              "rendered": Promise {},
              "type": "ul",
            },
          ],
          "type": "article",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
